routing/pathfind.go

```diff
@@ -188,6 +190,21 @@ func newRoute(sourceVertex route.Vertex,
 					"custom records")
 			}
 			customRecords = finalHop.records
+
+			// If we're attaching a payment addr but the receiver
+			// doesn't support both TLV and payment addrs, fail.
+			payAddr := supports(lnwire.PaymentAddrOptional)
+			if !payAddr && finalHop.paymentAddr != nil {
+				return nil, errors.New("cannot attach " +
+					"payment addr")
+			}
+
+			// Otherwise attach the mpp record if it exists.
+			if finalHop.paymentAddr != nil {
+				mpp = record.NewMPP(
```

Comment:
this logic should really have been in `newRoute` creation, but I guess we are stuck with doing it here now?

Reply:
this is in `newRoute`?

Reply:
lol